mlir/lib/Dialect/GPU/IR/GPUDialect.cpp

```diff
@@ -1782,6 +1787,12 @@ ParseResult GPUModuleOp::parse(OpAsmParser &parser, OperationState &result) {
     props.targets = targetsAttr;
   }
 
+  // Parse optional module constraints attributes
+  if (succeeded(parser.parseOptionalKeyword("constraints"))) {
+    props.constraints = cast<GPUModuleConstraintsAttr>(
+        GPUModuleConstraintsAttr::parse(parser, {}));
```

Comment:
This will cause issues if parsing fails. Use `parser.parseAttribute` intead ([see](https://github.com/llvm/llvm-project/blob/ba60d8a11af2cdd7e80e2fd968cdf52adcabf5a1/mlir/lib/Dialect/SPIRV/IR/SPIRVOps.cpp#L1459)).

Reply:
sorted